packages/domonda-query-params/__tests__/parseQueryParams.test.ts

```diff
@@ -236,3 +236,45 @@ it('should parse the complete model', () => {
     enum: 'VALID_ENUM',
   });
 });
+
+it('should not allow mutations on resulting parameters', () => {
+  interface State {
+    str: string;
+    num: number;
+    arr: string[];
+  }
+
+  const model: QueryModel<State> = {
+    str: {
+      type: 'string',
+      defaultValue: '',
+    },
+    num: {
+      type: 'number',
+      defaultValue: 0,
+    },
+    arr: {
+      type: 'array',
+      defaultValue: [''],
+    },
+  };
+
+  const state: State = {
+    str: 'one',
+    num: 1,
+    arr: ['one'],
+  };
+
+  const values = parseQueryParams(stringify(state, { prependQuestionMark: true }), model);
+
+  try {
+    values.str = 'two';
+    values.num = 2;
+    values.arr[0] = 'two';
+  } catch (err) {
```

Comment:
First object mutation (which is: `values.str = 'two';`) will throw immediately which means that the rest of the code wont even get reached.

Consider separating each mutation into its own `try-catch` block.